### PR TITLE
WebGPURenderer: Support `BatchedMesh`.

### PR DESCRIPTION
**Pull request by RenaudRohlinger**
Related issue: https://github.com/mrdoob/three.js/issues/22376

**Description**

This PR adds BatchMesh support to the WebGPURenderer on both backends. At this moment the WebGPU doesn't use any multi draw, let's add it once multiDrawIndirect is available (more info https://github.com/gpuweb/gpuweb/pull/2315).

On my MacBook Pro M1 Max:
WebGPU: 120fps+, 4ms CPU average, 5GPU
WebGL: 65 fps, 7.6ms CPU, 5GPU

Example:
https://raw.githack.com/renaudrohlinger/three.js/utsubo/feat/webgpu-batch/examples/webgpu_mesh_batch.html

Also fixed some issue regarding timestamps getting reset when computeAsync wasn't being used.
Finally, I refactored the `WebGLBackend` with `WebGLBufferRenderer`.

*This contribution is funded by [Utsubo](https://utsubo.com)*

### CODE REVIEW

**Comment by vegarringdal** [comment]
Nice

Tried the sample link, dont see much difference between webgpu/webgl.
Maybe I did something wrong ?
Added som images, ran both at the same time.

AMD RX5700XT
AMD Ryzen 7 3700X 8-Core Processor 3.59 GHz
Windows 11

Edge
image

Chrome
image

**Comment by RenaudRohlinger** [comment]
Which reminds me that maybe we should add the `GPURenderBundle` and consider the PR of @aardgoose. This PR will benefit a lot from it.

Related: 

https://github.com/mrdoob/three.js/issues/26876
https://github.com/mrdoob/three.js/pull/26983

/cc @sunag

File: examples/jsm/renderers/common/Renderer.js [hunk 1]
```diff
@@ -330,7 +330,7 @@ class Renderer {
 
 	}
 
-	_renderScene( scene, camera ) {
+	async _renderScene( scene, camera ) {
```
[the remark this answers is not in the document]
Comment (RenaudRohlinger) [reply examples/jsm/renderers/common/Renderer.js:333]: Ah yes, make sense!

**Comment by RenaudRohlinger** [comment]
@vegarringdal I guess 20k instances doesn't seems enough to see noticeable differences. My numbers comes from a sample of 40k instances. Also I don't want to burn the CPUs of developers so I guess 20k is a good limit.

File: examples/jsm/renderers/common/Renderer.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (RenaudRohlinger) [reply examples/jsm/renderers/common/Renderer.js:333]: Done 👍

**Comment by vegarringdal** [comment]
@RenaudRohlinger 

K, would nice to be able can compare performance with minimal work.
So if it will be an official sample, then it would prefer to be able to try and kill my cpu/gpu  😄 

But anyways, thanks for helping making the webgpu part more complete 👍 
Will be fun to see what's possible with webgpu/threejs in the future.

**Comment by RenaudRohlinger** [comment]
Will check! Probably just an issue with the example or the `BatchNode` not supporting dynamic update yet.

**Pull request merged**